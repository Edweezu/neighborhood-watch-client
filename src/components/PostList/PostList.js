import React from 'react'
import MainContext from '../../contexts/MainContext';
import { getPostsForCategory } from '../../helpers'
import Post from '../Post/Post'


class PostList extends React.Component {

    static contextType = MainContext

    render () {

        const { city_id, posts=[] } = this.context
        const { categoryid } = this.props
        const postList = getPostsForCategory(posts, categoryid, city_id) || []
        // console.log('posts', posts)
        // console.log('city_id', city_id)
        // console.log('categoryid', categoryid)
        // console.log('postlist', postList)

        return (
            <section className='PostList'>
                {postList.map(post => (
                    <li key={post.id}>
                        <Post 
                            id={post.id}
                            subject={post.subject}
                            message={post.message}
                            date_created={post.date_created}
                            category_id={post.category_id}
                            city_id={post.city_id}
                            user_id={post.user_id}
                        />
                    </li>
                ))}
            </section>
        )
    }
}

export default PostList