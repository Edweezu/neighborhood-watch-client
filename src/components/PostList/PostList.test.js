import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import PostList from './PostList'


describe('PostList component', () => {
    const props = {
        categoryId: 1
    }

    it('renders the given component', () => {
        const wrapper = shallow(<PostList />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it(`renders the component's given props`, () => {
        const wrapper = shallow(<PostList {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})