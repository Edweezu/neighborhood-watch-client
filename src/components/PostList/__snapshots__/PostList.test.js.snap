// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostList component renders the component's given props 1`] = `
<section
  className="PostList"
/>
`;

exports[`PostList component renders the given component 1`] = `
<section
  className="PostList"
/>
`;
